pkg/quota/v1/evaluator/core/persistent_volume_claims.go

```diff
@@ -159,23 +159,46 @@ func (p *pvcEvaluator) Usage(item runtime.Object) (corev1.ResourceList, error) {
 		result[storageClassClaim] = *(resource.NewQuantity(1, resource.DecimalSI))
 	}
 
-	// charge for storage
-	if request, found := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; found {
-		roundedRequest := request.DeepCopy()
+	requestedStorage := p.getStorageUsage(pvc)
+	if requestedStorage != nil {
+		result[corev1.ResourceRequestsStorage] = *requestedStorage
+		// charge usage to the storage class (if present)
+		if len(storageClassRef) > 0 {
+			storageClassStorage := corev1.ResourceName(storageClassRef + storageClassSuffix + string(corev1.ResourceRequestsStorage))
+			result[storageClassStorage] = *requestedStorage
+		}
+	}
+
+	return result, nil
+}
+
+func (p *pvcEvaluator) getStorageUsage(pvc *corev1.PersistentVolumeClaim) *resource.Quantity {
+	var result *resource.Quantity
+	roundUpFunc := func(i *resource.Quantity) *resource.Quantity {
+		roundedRequest := i.DeepCopy()
 		if !roundedRequest.RoundUp(0) {
 			// Ensure storage requests are counted as whole byte values, to pass resourcequota validation.
 			// See http://issue.k8s.io/94313
-			request = roundedRequest
+			return &roundedRequest
 		}
+		return i
+	}
 
-		result[corev1.ResourceRequestsStorage] = request
-		// charge usage to the storage class (if present)
-		if len(storageClassRef) > 0 {
-			storageClassStorage := corev1.ResourceName(storageClassRef + storageClassSuffix + string(corev1.ResourceRequestsStorage))
-			result[storageClassStorage] = request
+	if userRequest, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; ok {
+		result = roundUpFunc(&userRequest)
+	}
+
+	if len(pvc.Status.AllocatedResources) == 0 {
+		return result
+	}
+
+	// if AllocatedResources is set and is greater than user request, we should use it.
+	if allocatedRequest, ok := pvc.Status.AllocatedResources[corev1.ResourceStorage]; ok {
```

Comment:
should this be feature gated?

Reply:
may be - but that would mean if any reduction in PVC requests were performed and then feature gate got disabled, it will go back to reporting sort of invalid size in quota requests.

Reply:
actually we cover this in KEP:

> Yes the feature gate can be disabled once enabled. However quota resources present in the cluster will be based off a field(`pvc.Status.AllocatedResources`) which is no longer updated.
  Currently without this feature - quota calculation is entirely based on `pvc.Spec.Resources` and when feature is enabled it will based off `max(pvc.Spec.Resources, pvc.Status.AllocatedResources)`
  so when the feature is disabled, cluster might be reporting stale quota. To fix this issue - cluster admins can re-create `ResourceQuota` objects so as quota controller can recompute the
  quota using `pvc.Spec.Resources`.

Reply:
I think it's better if we base quota off of the feature gate. Yes, it means quota won't be guaranteed enforceable until the feature is GA and can't be turned off, but I think we want to follow rollback and feature disablement guidance here.